x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/assembler/ExecutionManager.java

```diff
@@ -177,7 +178,7 @@ public Executable assemble(
      */
     public Executable assemble(List<List<Attribute>> listOfKeys, List<PhysicalPlan> plans, Limit limit) {
         if (cfg.fetchSize() > SAMPLE_MAX_PAGE_SIZE) {
-            throw new EqlIllegalArgumentException("Fetch size cannot be greater than [{}]", SAMPLE_MAX_PAGE_SIZE);
+            throw new EqlClientException("Fetch size cannot be greater than [{}]", SAMPLE_MAX_PAGE_SIZE);
         }
```

Comment:
nit: maybe also `VerificationException` here.

Reply:
I've updated this usage, but with a newly introduced `QlClientIllegalArgumentException`. This is somewhat more fitting and this type of exception is required by the generic QL code changes (which this PR grew with).